tests/datasource_test.go

```diff
@@ -114,6 +138,72 @@ var _ = Describe("DataSource", func() {
 		_ = waitForReadyCondition(ds, corev1.ConditionFalse, "NoSource")
 	})
 
+	DescribeTable("[test_id:TODO] Labels should be copied to DataSource", func(sourceFn func() cdiv1.DataSourceSource, createSource func()) {
+		By("Create source for DataSource")
+		createSource()
+
+		By("Create DataSource")
+		ds := newDataSource(ds1Name)
+		ds.Spec.Source = sourceFn()
+		ds, err := f.CdiClient.CdiV1beta1().DataSources(ds.Namespace).Create(context.TODO(), ds, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		ds = waitForReadyCondition(ds, corev1.ConditionTrue, "Ready")
+
+		By("Check for labels on DataSource")
+		ds, err = f.CdiClient.CdiV1beta1().DataSources(ds.Namespace).Get(context.TODO(), ds.Name, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ds.Labels).To(HaveKeyWithValue(testKubevirtIoKey, testKubevirtIoValue))
+		Expect(ds.Labels).To(HaveKeyWithValue(testInstancetypeKubevirtIoKey, testInstancetypeKubevirtIoValue))
+		Expect(ds.Labels).To(HaveKeyWithValue(testKubevirtIoKeyExisting, testKubevirtIoNewValueExisting))
```

Comment:
I think you're not guaranteed to have the labels when reaching "Ready=true". You could use eventually here